src/main/python/mmlspark/lightgbm/LightGBMClassifier.py

```diff
@@ -34,33 +34,45 @@ def saveNativeModel(self, filename, overwrite=True):
     @staticmethod
     def loadNativeModelFromFile(filename, labelColName="label",
                                 featuresColName="features", predictionColName="prediction",
-                                probColName="probability", rawPredictionColName="rawPrediction"):
+                                probColName="probability", rawPredictionColName="rawPrediction",
+                                leafPredictionColName="leafPrediction",
+                                featuresShapColName="featuresShap"):
         """
         Load the model from a native LightGBM text file.
         """
         ctx = SparkContext._active_spark_context
         loader = ctx._jvm.com.microsoft.ml.spark.lightgbm.LightGBMClassificationModel
         java_model = loader.loadNativeModelFromFile(filename, labelColName,
                                                     featuresColName, predictionColName,
-                                                    probColName, rawPredictionColName)
+                                                    probColName, rawPredictionColName,
+                                                    leafPredictionColName, featuresShapColName)
         return JavaParams._from_java(java_model)
 
     @staticmethod
     def loadNativeModelFromString(model, labelColName="label",
                                   featuresColName="features", predictionColName="prediction",
-                                  probColName="probability", rawPredictionColName="rawPrediction"):
+                                  probColName="probability", rawPredictionColName="rawPrediction",
+                                  leafPredictionColName="leafPrediction",
+                                  featuresShapColName="featuresShap"):
         """
         Load the model from a native LightGBM model string.
         """
         ctx = SparkContext._active_spark_context
         loader = ctx._jvm.com.microsoft.ml.spark.lightgbm.LightGBMClassificationModel
         java_model = loader.loadNativeModelFromString(model, labelColName,
                                                       featuresColName, predictionColName,
-                                                      probColName, rawPredictionColName)
+                                                      probColName, rawPredictionColName,
+                                                      leafPredictionColName, featuresShapColName)
         return JavaParams._from_java(java_model)
 
     def getFeatureImportances(self, importance_type="split"):
         """
         Get the feature importances as a list.  The importance_type can be "split" or "gain".
         """
         return list(self._java_obj.getFeatureImportances(importance_type))
+
+    def getFeatureShaps(self, vector):
+        """
+        Get the local shap feature importances.
+        """
+        return list(self._java_obj.getFeatureShaps(vector))

```

Comment:
noticed this doesn't actually quite work as the scala API needs to take in an array instead of a vector, will fix it in a subsequent PR with another issue for multiclass SHAP values

Reply:
fixed it in this PR